tests/test_updater_with_simulator.py

```diff
@@ -110,8 +111,13 @@ def test_targets(self):
             with open(local_path, "rb") as f:
                 self.assertEqual(f.read(), content)
 
-        # TODO: run the same download tests for target paths like "dir/file2")
-        # This currently fails because issue #1576
+            if "/" in targetpath:
+                # assert local_path != targetpath because of the URL encoding
+                # make target_path absolute as local_path
+                target_path = os.path.join(self.targets_dir, targetpath)
+                self.assertNotEqual(target_path, local_path)
```

Comment:
I don't know if this test case makes sense, it is too specific and may not be correct when the api changes further. Can we at least move it to a separate test?

Reply:
I wanted to showcase that the given path is encoded and as a consequence changed if it includes a directory in it.
I decided to use an if statement in order to add as little as possible changes as the code used for the directory case is the same as the other two cases.

Reply:
Yes, I believe the test cases should be as short and easy to understand as possible and this deserves a separate test.